lib/pbench/server/api/resources/graphql_api.py

```diff
@@ -1,6 +1,254 @@
 import requests
+import datetime
+import json
 from flask_restful import Resource, abort
-from flask import request, make_response
+from flask import request, make_response, jsonify
+from pbench.server.api.resources.auth import auth
+from pbench.server.api.resources.models import MetadataModel
+from pbench.server.api.resources.database import Database
+
+
+class UserMetadata(Resource):
+    """
+    Abstracted pbench API for handling user metadata by using graphql schema
+    """
+
+    def __init__(self, config, logger):
+        self.server_config = config
+        self.logger = logger
+
+    @auth.login_required()
+    def post(self):
+        """
+        Post request for creating metadata instance for a user.
+        This requires a JWT auth token in the header field
+
+        This requires a JSON data with required user metadata fields
+        {
+            "config": "config",
+            "description": "description",
+        }
+
+        Required headers include
+            Authorization:   JWT token (user received upon login)
+
+        :return: JSON Payload
+            response_object = {
+                    "status": "success"
+                    "data" {
+                        "id": "metadata_id",
+                        "config": "Config string"
+                        "description": "Description string"
+                        }
+                }
+        """
+        post_data = request.get_json()
+        if not post_data:
+            self.logger.warning("Invalid json object: %s", request.url)
+            abort(400, message="Invalid json object in request")
+
+        config = post_data.get("config")
+        if not config:
+            self.logger.warning("Config not provided during metadata creation")
+            abort(400, message="Please provide a config string")
+
+        description = post_data.get("description")
+        if not description:
+            self.logger.warning("Description not provided during metadata creation")
+            abort(400, message="Please provide a description string")
+        current_user_id = auth.current_user().id
+
+        try:
+            # Create a new metadata session
+            metadata_session = MetadataModel(
+                created=str(datetime.datetime.now()),
+                config=config,
+                description=description,
+                user_id=current_user_id
+            )
+            # insert the metadata session for a user
+            Database.db_session.add(metadata_session)
+            Database.db_session.commit()
+            self.logger.info("New user metadata session created")
+        except Exception:
+            self.logger.exception("Exception occurred during Metadata creation")
+            abort(500, message="INTERNAL ERROR")
+        else:
+            response_object = {
+                "status": "success",
+                "data": {
```

Comment:
We should consider adding the `updated` field to the metadata model and return it with this response object as the field is used to visualize last modified values in the table view.